src/Http/Controllers/Admin/FileLibraryController.php

```diff
@@ -287,6 +289,43 @@ public function bulkUpdate()
         ], 200);
     }
 
+    /**
+     * @param Request $request
+     * @return string
+     */
+    public function getStaticLink(Request $request) {
+        $disk = $this->config->get('twill.media_library.disk');
+        $diskConfig = config('filesystems.disks.' . $disk);
+
+        $client = new S3Client([
+            'credentials' => [
+                'key'    => $diskConfig['key'],
+                'secret' => $diskConfig['secret']
+            ],
+            'region' => $diskConfig['region'],
+            'version' => 'latest'
+        ]);
+
+        // does this object have public-read?
+        $public = collect($client->getObjectAcl([
+            'Bucket' => $diskConfig['bucket'], // REQUIRED
+            'Key' => $request->uuid, // REQUIRED
+        ])->search('Grants'))
+            ->where('Permission','READ')
+            ->where('Grantee.URL','http://acs.amazonaws.com/groups/global/AllUsers')
+            ->isNotEmpty();
+
+        if(!$public) {
+            $client->putObjectAcl([
+                'ACL' => 'public-read',
+                'Bucket' => $diskConfig['bucket'], // REQUIRED
+                'Key' => $request->uuid, // REQUIRED
+            ]);
+        }
+
+        return Storage::disk($disk)->url($request->uuid);
+    }
+
```

Comment:
This looks like custom code you didn't intend to add to this PR. I'm curious though, what do you use this for?

Reply:
Yep I just removed this. We by default sign all file and media library URLs. This allows us to generate a static link that never expires by modifying the public access value on s3